### ISSUE TITLE
OroCommerce 4.1.2 UTF8MB4 Install error “1071 Specified key was too long; max key length is 3072 bytes”

### ISSUE THREAD

**Issue by chrisleup** [oroinc/orocommerce#114]
Fresh pull of the recent 4.1.2 package and when config.yml is set to use utf8mb4, during Database Setup/Process Migrations I get the following errors. However when installed using just utf8, the install succeeds properly.. I have used utf8mb4 with 4.1.0 and 4.1.1 as well as the entire 3.1 branch with the current setup perfectly. This was just introduced in the recent revision. There's lots in google about VARCHAR(191) for utf8mb4 but thats's with a limit of 767 bytes(old mysql limit), not 3072 as is shown below.

Can anybody else reproduce this?
```
> Oro\Bundle\DataAuditBundle\Migrations\Schema\OroDataAuditBundleInstaller
    ERROR: An exception occurred while executing 'CREATE TABLE oro_audit (id INT AUTO_INCREMENT NOT NULL, user_id INT DEFAULT NULL, organization_id INT DEFAULT NULL, impersonation_id INT DEFAULT NULL, action VARCHAR(8) DEFAULT NULL, logged_at DATETIME DEFAULT NULL COMMENT '(DC2Type:datetime)', object_id VARCHAR(255) DEFAULT NULL, object_class VARCHAR(255) NOT NULL, object_name VARCHAR(255) DEFAULT NULL, version INT DEFAULT NULL, type VARCHAR(255) NOT NULL, transaction_id VARCHAR(255) NOT NULL, owner_description VARCHAR(255) DEFAULT NULL, additional_fields LONGTEXT DEFAULT NULL COMMENT '(DC2Type:array)', INDEX IDX_5FBA427CA76ED395 (user_id), INDEX idx_oro_audit_type (type), UNIQUE INDEX idx_oro_audit_version (object_id, object_class, version, type), UNIQUE INDEX idx_oro_audit_transaction (object_id, object_class, transaction_id, type), INDEX idx_oro_audit_logged_at (logged_at), INDEX idx_oro_audit_object_class (object_class), INDEX idx_oro_audit_obj_by_type (object_id, object_class, type), INDEX idx_oro_audit_owner_descr (owner_description), INDEX idx_oro_audit_organization_id (organization_id), INDEX IDX_5FBA427C26F87DB8 (impersonation_id), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci ENGINE = InnoDB':

SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes
```
Also posted at [Stack Overflow](https://stackoverflow.com/questions/61195090/orocommerce-4-1-2-utf8mb4-install-error-1071-specified-key-was-too-long-max-ke)

**Comment by mbessolov** [comment]
Please use Slack or SO when asking general questions. No need to duplicate questions here.
If you want to report a bug, please follow the template and provide full STR and detailed information about the environment where the problem occurs.

**Comment by chrisleup** [comment]
Thanks Michael, wasn't sure how you wanted to track it as it does appear to be a bug as I've tested against MySQL 5.7.29 and MariaDB 10.4.12 . =(

**Comment by jimohalloran** [comment]
On @chrisleup 's [SO question](https://stackoverflow.com/questions/61195090/orocommerce-4-1-2-utf8mb4-install-error-1071-specified-key-was-too-long-max-ke) @anyt confirmed this has been logged as a bug with internal id BAP-18992.  What's the status of this issue?  I've got the same problem on OroCRM 4.1.2 (it's a platform issue, so it makes sense both OroCRM and OroCommerce would be affected).

**Comment by chrisleup** [comment]
I followed up with an issue on GitHub and Michael Bessolov has noted it's
been fixed as of verison c2ceac2 .. Should also be included in the next
4.1.x release.
https://github.com/oroinc/platform/issues/1013
